test/e2e/pkg/manifest.go

```diff
@@ -88,6 +88,17 @@ type ManifestNode struct {
 	// on the machine where the test is being run.
 	Version string `toml:"version"`
 
+	// BuiltinProxyMode specifies the type of proxy we should use if the
+	// testnet's ABCIProtocol is set to "builtin". This can either be "sync",
+	// which uses [proxy.NewLocalClientCreator] to create ABCI client
+	// interfaces with a global mutex across all of them, or "unsync", which
+	// uses [proxy.NewUnsyncLocalClientCreator] to create ABCI client
+	// interfaces with a mutex per client.
+	//
+	// The "unsync" version effectively simulates the same concurrency model as
+	// the socket-based connection.
+	BuiltinProxyMode string `toml:"builtin_proxy_mode"`
```

Comment:
With the current patch, `ABCIProtocol` is set for the overall testnet, whereas `BuiltinProxyMode` is set per node.
Wouldn't it be simpler if `BuiltinProxyMode` was defined at the same level as `ABCIProtocol`? At the end of the day, there are related

Reply:
The original intention here was to have a mix of nodes with different concurrency models iff `ABCIProtocol` is set to `builtin` for the overall testnet.

This is chosen randomly here: https://github.com/tendermint/tendermint/pull/9830/files#diff-40ef656313ea7873bfaf38026d93c96559340552d918492dd3f68c723c04addfR221

Reply:
Yes yes, my question was rather if we want that extra flexibility in `BuiltinProxyMode` since we don't have it in `ABCIProtocol`. But this is a non-blocking comment... more of an open question

Reply:
Done!